### PR TITLE
Support BinderChannelBuilder.forTarget.

### PR DESCRIPTION
**Pull request by markb74**
Allows BinderChannel to be used with name resolvers.

### CODE REVIEW

File: testing/src/main/java/io/grpc/testing/FakeNameResolverProvider.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/8636")
+public final class FakeNameResolverProvider extends NameResolverProvider {
+
+  /**
+   * Register a new resolver.
+   *
+   * @param targetUri The URI to resolve when requested.
+   * @param address The address to return for the target URI.
+   */
+  public static final void register(String targetUri, SocketAddress address) {
```
Comment (ejona86) [inline testing/src/main/java/io/grpc/testing/FakeNameResolverProvider.java:39]: I think we should delete this method and instead expose the constructor. This method is a bad pattern. After a test using this is over, it should really deregister the provider. Conceptually (but maybe using `@After`):

```
provider = new FakeNameResolverProvider(URI.create(targetUri), address);
NameResolverRegistry.getDefaultRegistry().register(provider);
try {
  ...
} finally {
  NameResolverRegistry.getDefaultRegistry().deregister(provider);
}
```

I don't think this class needs to make that try-finally _easier_, but it shouldn't encourage being used in a register-only fashion.

File: testing/src/main/java/io/grpc/testing/FakeNameResolverProvider.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.NameResolver;
+import io.grpc.NameResolverProvider;
+import io.grpc.NameResolverRegistry;
+import io.grpc.Status;
+import java.net.SocketAddress;
+import java.net.URI;
+
+/** A name resolver to always resolve the given URI into the given address. */
+public final class FakeNameResolverProvider extends NameResolverProvider {
```
[the remark this answers is not in the document]
Comment (ejona86) [reply testing/src/main/java/io/grpc/testing/FakeNameResolverProvider.java:31]: We may want to discuss when you used this internally, as I expect it was probably with a grpc-maintainer hat on (e.g., when writing a transport). It wouldn't be "against the rules" to have this be internal but let some specific tests access it (I expect we're doing that already).

**Review by ejona86 — APPROVED** [review]

File: testing/src/main/java/io/grpc/testing/FakeNameResolverProvider.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (markb74) [reply testing/src/main/java/io/grpc/testing/FakeNameResolverProvider.java:31]: That's a good point, on consideration moving to `io.grpc.internal.testing`.

File: testing/src/main/java/io/grpc/testing/FakeNameResolverProvider.java (hunk 1 above)
Comment (markb74) [reply testing/src/main/java/io/grpc/testing/FakeNameResolverProvider.java]: Good point, fixed.

**Pull request merged**